clients/instance/ibm-pi-snapshot.go

```diff
@@ -81,12 +81,29 @@ func (f *IBMPISnapshotClient) GetAll() (*models.Snapshots, error) {
 	return resp.Payload, nil
 }
 
-// Create or Restore a Snapshot
-func (f *IBMPISnapshotClient) Create(instanceID, snapshotID, restoreFailAction string) (*models.Snapshot, error) {
+// Restore a Snapshot
+func (f *IBMPISnapshotClient) Restore(instanceID, snapshotID, restoreFailAction string, body *models.SnapshotRestore) (*models.Snapshot, error) {
 	params := p_cloud_p_vm_instances.NewPcloudPvminstancesSnapshotsRestorePostParams().
 		WithContext(f.ctx).WithTimeout(helpers.PICreateTimeOut).
 		WithCloudInstanceID(f.cloudInstanceID).WithPvmInstanceID(instanceID).
-		WithSnapshotID(snapshotID).WithRestoreFailAction(&restoreFailAction)
+		WithSnapshotID(snapshotID).WithRestoreFailAction(&restoreFailAction).
```

Comment:
I did not mean to duplicate the code.
Just call Create() from here or vice versa.

Reply:
The existing create which does a restore does not call the restore with all the proper parameters. You currently cannot pass a fail action to it.

Reply:
I see, if `restoreFailAction` is required then we can skip the call.
If Create() itself is failing then we need to fix it.